android/hybridcommon/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -231,7 +240,26 @@ private fun validatePurchaseParams(
     }
 }
 
-@Suppress("LongParameterList", "LongMethod", "NestedBlockDepth")
+@Suppress("ReturnCount")
+private fun validateAddOnStoreProducts(
+    addOnStoreProductsOption: Any?,
+): List<PurchasableItem.Product>? {
+    if (addOnStoreProductsOption == null) {
+        return null
+    }
+    val addOnStoreProductsList = addOnStoreProductsOption as? List<*> ?: return null
+    val parsedProducts = addOnStoreProductsList.mapNotNull { item ->
+        when (item) {
+            is PurchasableItem.Product -> item
+            is Map<*, *> -> PurchasableItem.Product.fromMap(map = item)
+            else -> return null
+        }
+    }
+
+    return parsedProducts
+}
+
+@Suppress("LongParameterList")
```

Comment:
Here's what's going on with these function definitions: I wanted to avoid making `PurchasableItem` public and I only want to support purchasing add-ons in the `purchase(options)` function. So, I created purchase functions without the add-on params to maintain the old interfaces. These functions then call purchase functions with the add-on params that work the same, except that they attach the add-ons when provided.